### PR TITLE
update to context datastores

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

File: testutil/fakegraphsync.go [hunk 1]
```diff
@@ -422,6 +422,11 @@ func (fgs *FakeGraphSync) Stats() graphsync.Stats {
 	return graphsync.Stats{}
 }
 
+func (fgs *FakeGraphSync) RegisterOutgoingRequestProcessingListener(graphsync.OnOutgoingRequestProcessingListener) graphsync.UnregisterHookFunc {
```
Comment (arajasek) [inline testutil/fakegraphsync.go:425]: where this came from why this exist

**Review by arajasek — APPROVED** [review]

**Comment by arajasek** [comment]
Updated to latest go-ds-versioning, but test seems to be flaky failing (passes locally)

File: testutil/fakegraphsync.go (hunk 1 above)
Comment (hannahhoward) [reply testutil/fakegraphsync.go:425]: yea that's a mock GraphExchange interface that needs a new method cause Graphsync updated and added it.. I'm moving away from putting methods on GraphExchange and towards only implementing them on the main implementation to simplify this. But later.

**Comment by hannahhoward** [comment]
@arajasek go-data-transfer I believe still suffers from unreliable CI -- I haven't been in it in a bit but @dirkmc can confirm.

**Review by hannahhoward — APPROVED** [review]
LGTM

**Pull request merged**